### DISCUSSION TITLE
Missing key value in the case of simple key/value pairing

### DISCUSSION THREAD

**Discussion by Thisisjuke** [Thinkmill/keystatic#281] [Q&A] [answered]
I wanted to do a simple key value to handle common error strings inside my forms: 
![image](https://user-images.githubusercontent.com/23522065/237063192-c03362e8-aca9-4e5a-a6bc-2231fdebdcdd.png)

`keystatic.config.js`
```js
formCommon: collection({
            label: 'Form Common Texts',
            path: 'src/modules/keystatic/content/form-common/*/',
            slugField: 'key',
            schema: {
                key: fields.text({
                    label: 'Key',
                    validation: { length: { min: 1 } },
                }),
                value: fields.text({
                    label: 'Value',
                    validation: { length: { min: 1 } },
                }),
            },
        }),
```

When log the content of my data, I have :
![image](https://github.com/Thinkmill/keystatic/assets/23522065/d50cdcc9-4756-4e8e-a5cd-91ec7de90288)

I updated my `package.json` to:
```
    "@keystatic/core": "^0.0.98",
    "@keystatic/next": "^0.0.7",
```

Now, I have to do this:
```js
const col = await reader.collections[resource].read(slug)
if (col && 'key' in col && col?.key === null) {
    col.key = slug
}
return col
```

It's easy to remap it because I just asked `.read(slug)` .

## Expected Behavior

I want to have the slug by default when I ask for a `key` + `value` object so I don't have to set the key (slug) by myself.

## Current Behavior

Slug is `null`.

**Comment by simonswiss** [comment, marked as answer]
Hey!

The `slug` is not part of the data generated by Keystatic — since it's effectively stored as the file/folder name.

A common approach here when gathering data for a collection is to "augment" the object with something that stores the `slug`, if you need it:

```js
const col = await reader.collections[resource].read(slug)
return { ...col, key: slug }
```

That way the `key` property will be available alongside the rest of the data.

Hope it helps!
